src/main/java/net/sf/jabref/logic/integrity/IntegrityCheck.java

```diff
@@ -306,8 +306,9 @@ private BracketChecker(String field) {
     private static class YearChecker implements Checker {
 
         private static final Predicate<String> CONTAINS_FOUR_DIGIT = Pattern.compile("([^0-9]|^)[0-9]{4}([^0-9]|$)").asPredicate();
-        private static final Predicate<String> ENDS_WITH_FOUR_DIGIT = Pattern.compile("[0-9][0-9][0-9][0-9]$")
+        private static final Predicate<String> ENDS_WITH_FOUR_DIGIT = Pattern.compile("[0-9]{4}$")
                 .asPredicate();
+        private static final String regex = "[](){},.;!?<>%&$]";
```

Comment:
Please uppercase constant name and call it something like `PUNCTUATION_MARKS`. Just calling it regex doesn't really help in understanding what it's for.